Flaky TableNameRegistryTest.testConcurrentReadWriteAndReload

### Describe the bug

Failed on mac-cairo
```
2023-01-24T15:36:12.5582350Z 2023-01-24T15:36:11.335721Z I i.q.c.w.TableNameRegistryTest random seeds: 330270967725L, 1674574571335L
...
2023-01-24T15:36:16.6607340Z 2023-01-24T15:36:16.230234Z E i.q.TestListener ***** Test Failed *****io.questdb.cairo.wal.TableNameRegistryTest.testConcurrentReadWriteAndReload : 
2023-01-24T15:36:16.6608290Z java.lang.AssertionError: expected:<205> but was:<206>
2023-01-24T15:36:16.6608960Z 	at org.junit.Assert.fail(Assert.java:89)
2023-01-24T15:36:16.6609650Z 	at org.junit.Assert.failNotEquals(Assert.java:835)
2023-01-24T15:36:16.6610360Z 	at org.junit.Assert.assertEquals(Assert.java:647)
2023-01-24T15:36:16.6611470Z 	at org.junit.Assert.assertEquals(Assert.java:633)
2023-01-24T15:36:16.6613170Z 	at io.questdb.cairo.wal.TableNameRegistryTest.lambda$testConcurrentReadWriteAndReload$6(TableNameRegistryTest.java:283)
2023-01-24T15:36:16.6614630Z 	at io.questdb.cairo.AbstractCairoTest.lambda$assertMemoryLeak$4(AbstractCairoTest.java:276)
2023-01-24T15:36:16.6616290Z 	at io.questdb.test.tools.TestUtils.assertMemoryLeak(TestUtils.java:541)
2023-01-24T15:36:16.6618820Z 	at io.questdb.cairo.AbstractCairoTest.assertMemoryLeak(AbstractCairoTest.java:273)
2023-01-24T15:36:16.6619840Z 	at io.questdb.cairo.AbstractCairoTest.assertMemoryLeak(AbstractCairoTest.java:267)
2023-01-24T15:36:16.6620940Z 	at io.questdb.cairo.wal.TableNameRegistryTest.testConcurrentReadWriteAndReload(TableNameRegistryTest.java:212)
2023-01-24T15:36:16.6621840Z 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2023-01-24T15:36:16.6622680Z 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2023-01-24T15:36:16.6623630Z 	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2023-01-24T15:36:16.6624480Z 	at java.lang.reflect.Method.invoke(Method.java:566)
2023-01-24T15:36:16.6625290Z 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
2023-01-24T15:36:16.6626210Z 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
2023-01-24T15:36:16.6627110Z 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
2023-01-24T15:36:16.6628000Z 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
2023-01-24T15:36:16.6628860Z 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
2023-01-24T15:36:16.6629720Z 	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
2023-01-24T15:36:16.6630650Z 	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
2023-01-24T15:36:16.6631390Z 	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
2023-01-24T15:36:16.6632250Z 	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
2023-01-24T15:36:16.6633230Z 	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
2023-01-24T15:36:16.6634110Z 	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
2023-01-24T15:36:16.6634790Z 	at java.lang.Thread.run(Thread.java:829)
```

[testConcurrentReadWriteAndReload.zip](https://github.com/questdb/questdb/files/10496873/testConcurrentReadWriteAndReload.zip)


### To reproduce

_No response_

### Expected Behavior

_No response_

### Environment

```markdown
- **QuestDB version**:
- **OS**:
- **Browser**:
```


### Additional context

_No response_

--- a/core/src/main/java/io/questdb/cairo/TableNameRegistryFileStore.java
+++ b/core/src/main/java/io/questdb/cairo/TableNameRegistryFileStore.java
@@ -213,19 +213,19 @@ public class TableNameRegistryFileStore implements Closeable {
             if (fileLen > 4) {
                 int charLen = ff.readNonNegativeInt(fd, 0);
                 if (charLen * 2L + 4 != fileLen - 1) {
-                    LOG.error().$("invalid table name file [path=").$(path).$(", headerLen=").$(charLen).$(", fileLne=").$(fileLen).I$();
+                    LOG.error().$("invalid table name file [path=").$(path).$(", headerLen=").$(charLen).$(", fileLen=").$(fileLen).I$();
                     return null;
                 }
 
                 tableNameRoMemory.of(ff, path, fileLen, fileLen, MemoryTag.MMAP_DEFAULT);
                 String value = Chars.toString(tableNameRoMemory.getStr(0));
-                tableNameRoMemory.close();
                 return value;
             } else {
                 LOG.error().$("invalid table name file [path=").$(path).$(", fileLne=").$(fileLen).I$();
                 return null;
             }
         } finally {
+            tableNameRoMemory.close();
             ff.close(fd);
         }
     }
@@ -392,8 +392,8 @@ public class TableNameRegistryFileStore implements Closeable {
             }
 
             if (deletedRecordsFound > 0) {
-                LOG.info().$("compacting tables file [path=").$(path).$(']').$();
                 path.trimTo(pathRootLen);
+                LOG.info().$("compacting tables file").$();
                 compactTableNameFile(nameTableTokenMap, lastFileVersion, ff, path, currentOffset);
             } else {
                 tableNameMemory.jumpTo(currentOffset);

--- a/core/src/test/java/io/questdb/cairo/wal/TableNameRegistryTest.java
+++ b/core/src/test/java/io/questdb/cairo/wal/TableNameRegistryTest.java
@@ -270,8 +270,14 @@ public class TableNameRegistryTest extends AbstractCairoTest {
                         addedTables.remove(tableId);
 
                         // Retry remove table folder, until success, if table folder not clearly removed, reload may pick it up
+                        // Remove _txn file first
+                        rmPath.trimTo(configuration.getRoot().length()).concat(tableName).concat(TableUtils.TXN_FILE_NAME).$();
+                        configuration.getFilesFacade().remove(rmPath);
+
+                        // Remove table directory
                         for (int i = 0;
-                             i < 1000 && configuration.getFilesFacade().rmdir(rmPath.trimTo(configuration.getRoot().length()).concat(tableName).$()) != 0; i++) {
+                             i < 1000 && configuration.getFilesFacade().rmdir(rmPath.trimTo(configuration.getRoot().length()).concat(tableName).$()) != 0;
+                             i++) {
                             Os.pause();
                         }
                     }

